### PR TITLE
Fix to Hashicorp secret store vault getSecret error

### PR DESCRIPTION
**Pull request by shivanisinghnitp**
Signed-off-by: Shivani Singh <shivasingh@microsoft.com>

# Description

For the issue 4779 -> default value of vaultKVUsePrefix is true, due to which, BulkGetSecret method can't be used until it's set to false. Instead of having the user change that in the YAML, if we by default keep the value to be false, it would not cause issues such as 4779 to the users

## Issue reference

#4779
https://github.com/dapr/dapr/issues/4779

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: secretstores/hashicorp/vault/vault.go [hunk 1]
```diff
@@ -166,7 +166,7 @@ func (v *vaultSecretStore) Init(metadata secretstores.Metadata) error {
 		}
 	}
 
-	if !convertedVaultKVUsePrefix {
+	if convertedVaultKVUsePrefix {
```
Comment (pravinpushkar) [inline secretstores/hashicorp/vault/vault.go:169]: @shivanisinghnitp this change looks similar to the existing one. Is default value of `convertedVaultKVUsePrefix` being read somewhere else. Please make sure this is tested out.

**Review by akhilac1 — CHANGES_REQUESTED** [review]
@shivanisinghnitp  - Hashicorp vault documentation states the follows - 
'The path prefix (or alternatively the -mount flag for vault kv commands) tells Vault which secrets engine to which it should route traffic. When a request comes to Vault, it matches the initial path part using a longest prefix match and then passes the request to the corresponding secrets engine enabled at that path. Vault presents these secrets engines similar to a filesystem.'

The existing default of 'true' is appropriate considering the way Hashicopr Vault is designed. Also, the referenced issue does not discuss any limitations related to this attribute. 

Default is also clearly documented in dapr docs - https://docs.dapr.io/reference/components-reference/supported-secret-stores/hashicorp-vault/

I do not see this as a right approach to reset the default to false.Please revisit the proposed PR.

**Comment by berndverst** [comment]
I agree with @akhilac1. Also this would be a breaking change for several users of this component today.

I do not agree with changing the default value.

Surely, there is a different way to make `BulkGetSecret` work -- perhaps the implementation of that function must change.

File: secretstores/hashicorp/vault/vault.go (hunk 1 above)
Comment (berndverst) [reply secretstores/hashicorp/vault/vault.go:169]: Should this line remain unchanged?

**Review by berndverst — CHANGES_REQUESTED** [review]
On behalf of Contrib maintainers I do not approve of this change.

`BulkGetSecret` can be used to fetch all secrets with the `dapr` prefix by default. That is the correct behavior. If you wish to not use a prefix you need to configure the component to disable the prefix.

By default the `dapr` prefix (or provided prefix) should be continued to be used unless use of prefix is explicitly disabled.

cc @yaron2

**Pull request closed without merging**